worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java

```diff
@@ -237,25 +271,76 @@ private Extent wrapExtent(Extent extent, EventBus eventBus, EditSessionEvent eve
 
     // pkg private for TracedEditSession only, may later become public API
     boolean commitRequired() {
-        if (isQueueEnabled() && reorderExtent.commitRequired()) {
+        if (reorderExtent != null && reorderExtent.commitRequired()) {
             return true;
         }
-        if (isBatchingChunks() && chunkBatchingExtent.commitRequired()) {
+        if (chunkBatchingExtent != null && chunkBatchingExtent.commitRequired()) {
+            return true;
+        }
+        if (fastModeExtent != null && fastModeExtent.commitRequired()) {
             return true;
         }
         return false;
     }
 
     /**
      * Turns on specific features for a normal WorldEdit session, such as
-     * {@link #enableQueue() queuing} and {@link #setBatchingChunks(boolean)
+     * {@link #setReorderMode(ReorderMode)} reordering} and {@link #setBatchingChunks(boolean)
      * chunk batching}.
      */
     public void enableStandardMode() {
-        enableQueue();
+        setReorderMode(ReorderMode.MULTI_STAGE);
         setBatchingChunks(true);
     }
 
+    /**
+     * Sets the {@link ReorderMode} of this EditSession, and flushes the session.
+     *
+     * @param reorderMode The reorder mode
+     */
+    public void setReorderMode(ReorderMode reorderMode) {
+        if (reorderMode == ReorderMode.FAST && fastModeExtent == null) {
+            throw new IllegalArgumentException("An EditSession without a fast mode tried to use it for reordering!");
+        }
+        if (reorderMode == ReorderMode.MULTI_STAGE && reorderExtent == null) {
+            throw new IllegalArgumentException("An EditSession without a reorder extent tried to use it for reordering!");
+        }
+        flushSession();
```

Comment:
Surround this in `commitRequired()` like the other `flushSession()` call? Or maybe we should just make `flushSession()` do the check?